pkg/virt-controller/watch/snapshot/snapshot.go

```diff
@@ -686,24 +688,66 @@ func (ctrl *VMSnapshotController) getVolumeStorageClassForVolume(namespace strin
 		return ctrl.getVolumeSnapshotClass(*pvc.Spec.StorageClassName)
 	}
 	if volume.VolumeSource.DataVolume != nil {
-		pvcKey := cacheKeyFunc(namespace, volume.VolumeSource.DataVolume.Name)
-		// TODO Change when PVC rename PR is implemented
-
-		obj, exists, err := ctrl.PVCInformer.GetStore().GetByKey(pvcKey)
+		storageClassName, err := ctrl.getStorageClassNameForDV(namespace, volume.VolumeSource.DataVolume.Name)
 		if err != nil {
 			return "", err
 		}
 
-		if !exists {
-			log.Log.V(3).Infof("PVC not in cache [%s]", pvcKey)
-			return "", fmt.Errorf("PVC for the DataVolume not found")
+		return ctrl.getVolumeSnapshotClass(storageClassName)
+	}
+
+	return "", fmt.Errorf("Volume type does not suport snapshots")
+}
+
+func (ctrl *VMSnapshotController) getStorageClassNameForDV(namespace string, dvName string) (string, error) {
+	// First, look up DV's StorageClass
+	key := cacheKeyFunc(namespace, dvName)
+
+	obj, exists, err := ctrl.DVInformer.GetStore().GetByKey(key)
+	if err != nil {
+		return "", err
+	}
+
+	var dv *cdiv1.DataVolume
+	if exists {
+		dv = obj.(*cdiv1.DataVolume)
+		if dv.Spec.PVC == nil || dv.Spec.PVC.StorageClassName == nil {
+			log.Log.V(3).Infof("DV [%s] not found", key)
+			return "", fmt.Errorf("DV '%s' not found", key)
```

Comment:
Could you explain what this `not found` means? It looks to me like you found it in this code branch.

Reply:
nil `dv.Spec.PVC.StorageClassName` is okay here.  Either means use the default storage class or no storage class (statically allocated PVs)

Reply:
This returned error still looks suspicious. As far as I can read the code, the PVC may not be found, but it is still misleading, because why would a DV or PVC not be found if there is no StorageClassName set?

Reply:
You're right, this is misleading. Refactoring...